fix(linter): fix `jsx-a11y/label-has-associated-control` default values

Sysix opened the pull request:
`LabelHasAssociatedControl.from_configuration` never got called. 
Only the `LabelHasAssociatedControlConfig::default()`.

closes #11644

Sysix commented on apps/oxlint/fixtures/issue_11644/.oxlintrc.json:
```diff
@@ -0,0 +1,26 @@
+    "typescript/no-non-null-assertion": 0,
+    "unicorn/filename-case": [2, { "case": "kebabCase" }],
+    "unicorn/no-nested-ternary": 0,
+    // "jsx-a11y/label-has-associated-control": ["warn", { "depth": 0}]
```
In `main`: setting to "warn" or "error" for this rule, will remove the diagnostics. Keeping it out commented will report it

camc314 commented:
### Merge activity

* **Jun 23, 8:56 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

Sysix committed:
fix(linter): fix `jsx-a11y/label-has-associated-control` default values (#11832)

`LabelHasAssociatedControl.from_configuration` never got called.
Only the `LabelHasAssociatedControlConfig::default()`.

closes #11644
